cirq-core/cirq/testing/repr_pretty_tester.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Any
+
+
+class FakePrinter:
+    """A fake of iPython's PrettyPrinter which captures text added to this printer.
+
+    Can be used in tests to test a classes `_repr_pretty_` method:
+
+        ```
+        p = FakePrinter()
+        s = object_under_test._repr_pretty(p, cycle=False)
+        assert p.text_pretty == 'my pretty text'
+        ```
```

Comment:
Nit: prefer to write like you'd see in the idle (remove backticks and indentation and add >>>)

\>\>\> p = FakePrinter()
\>\>\> s = object_under_test._repr_pretty(p, cycle=False)
\>\>\> p.text_pretty
'my pretty text'